### PR TITLE
Fix(eos_designs): Add redistribute connected under BGP for VRF default if no underlay

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add redistribute connected under BGP for VRF default if no underlay

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

In the corner case where we build BGP config for VRF default as part of network services (but not for underlay/overlay), we were missing "redistribute connected".

There will be a PR later to provide control for redistribute connected.

## How to test

Existing molecule exposed the issue and show the fix.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM

**Comment by ClausHolbechArista** [comment]
@carlbuchmann I don't think this will apply to 4.x since we did not support the corner case we hit here. It is part of the new BGP behavior when not using overlays.

File: python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py [hunk 1]
```diff
@@ -191,6 +191,12 @@ def _router_bgp_vrfs(self: AvdStructuredConfigNetworkServices) -> dict:
                     # Resetting bgp_vrf so we only add global keys if there are any neighbors for VRF default
                     bgp_vrf = {}
 
+                if vrf_name == "default" and self.shared_utils.underlay_routing_protocol == "none":
+                    # We need to add redistribute connected for the default VRF when underlay_routing_protocol is "none"
+                    bgp_vrf |= {
+                        "redistribute_routes": [{"source_protocol": "connected"}],
+                    }
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py]:
```suggestion
                    bgp_vrf["redistribute_routes"] = [{"source_protocol": "connected"}]
```

File: python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py [hunk 2]
```diff
@@ -191,6 +191,12 @@ def _router_bgp_vrfs(self: AvdStructuredConfigNetworkServices) -> dict:
                     # Resetting bgp_vrf so we only add global keys if there are any neighbors for VRF default
                     bgp_vrf = {}
 
+                if vrf_name == "default" and self.shared_utils.underlay_routing_protocol == "none":
```
Comment (gmuloc) [inline python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py]: Could we move this up to line 180 with an `else` (as we know it is VRF default), then add a check for `/ if bgp_vrf is not empty and 2/ routing protocol==none

This saves one comparison in this if statement and groups everything happening for VRF default in one block (which we could move toa function

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py (hunk 2 above)
Comment (ClausHolbechArista) [reply python-avd/pyavd/_eos_designs/structured_config/network_services/router_bgp.py]: done

**Review by gmuloc — APPROVED** [review]

**Pull request merged**